homeassistant/helpers/selector.py

```diff
@@ -82,72 +83,28 @@ def serialize(self) -> dict[str, dict[str, _T]]:
 
 
 @cache
-def _entity_features() -> dict[str, type[IntFlag]]:
-    """Return a cached lookup of entity feature enums."""
-    # pylint: disable=import-outside-toplevel
-    from homeassistant.components.alarm_control_panel import (
-        AlarmControlPanelEntityFeature,
-    )
-    from homeassistant.components.calendar import CalendarEntityFeature
-    from homeassistant.components.camera import CameraEntityFeature
-    from homeassistant.components.climate import ClimateEntityFeature
-    from homeassistant.components.cover import CoverEntityFeature
-    from homeassistant.components.fan import FanEntityFeature
-    from homeassistant.components.humidifier import HumidifierEntityFeature
-    from homeassistant.components.lawn_mower import LawnMowerEntityFeature
-    from homeassistant.components.light import LightEntityFeature
-    from homeassistant.components.lock import LockEntityFeature
-    from homeassistant.components.media_player import MediaPlayerEntityFeature
-    from homeassistant.components.notify import NotifyEntityFeature
-    from homeassistant.components.remote import RemoteEntityFeature
-    from homeassistant.components.siren import SirenEntityFeature
-    from homeassistant.components.todo import TodoListEntityFeature
-    from homeassistant.components.update import UpdateEntityFeature
-    from homeassistant.components.vacuum import VacuumEntityFeature
-    from homeassistant.components.valve import ValveEntityFeature
-    from homeassistant.components.water_heater import WaterHeaterEntityFeature
-    from homeassistant.components.weather import WeatherEntityFeature
-
-    return {
-        "AlarmControlPanelEntityFeature": AlarmControlPanelEntityFeature,
-        "CalendarEntityFeature": CalendarEntityFeature,
-        "CameraEntityFeature": CameraEntityFeature,
-        "ClimateEntityFeature": ClimateEntityFeature,
-        "CoverEntityFeature": CoverEntityFeature,
-        "FanEntityFeature": FanEntityFeature,
-        "HumidifierEntityFeature": HumidifierEntityFeature,
-        "LawnMowerEntityFeature": LawnMowerEntityFeature,
-        "LightEntityFeature": LightEntityFeature,
-        "LockEntityFeature": LockEntityFeature,
-        "MediaPlayerEntityFeature": MediaPlayerEntityFeature,
-        "NotifyEntityFeature": NotifyEntityFeature,
-        "RemoteEntityFeature": RemoteEntityFeature,
-        "SirenEntityFeature": SirenEntityFeature,
-        "TodoListEntityFeature": TodoListEntityFeature,
-        "UpdateEntityFeature": UpdateEntityFeature,
-        "VacuumEntityFeature": VacuumEntityFeature,
-        "ValveEntityFeature": ValveEntityFeature,
-        "WaterHeaterEntityFeature": WaterHeaterEntityFeature,
-        "WeatherEntityFeature": WeatherEntityFeature,
-    }
+def _entity_feature_flag(domain: str, enum_name: str, feature_name: str) -> int:
+    """Return a cached lookup of an entity feature enum."""
+    module = importlib.import_module(f"homeassistant.components.{domain}")
```

Comment:
If its not already in `sys.modules` and it runs in the event loop, this will do blocking I/O.

Reply:
My impression was that was already happening, is that right? Or is blocking loading the components on startup somehow better?

Reply:
It might be that they are all already imported before the event loop starts so it's not a problem. If so they will already be in sys.modules

Reply:
In the example in https://github.com/home-assistant/core/issues/114318 it's calling `_load_services_file` to load the services file off of disk which appears to happen from a background thread, and there it is loading the schema validation and then the stream component and numpy etc. The `_SERVICES_SCHEMA` is only ever loaded from `_load_services_file` so it seems to work out.

Let me know if there are any conventions/naming for baking in this assumption about the current behavior to make it more understandable.

Reply:
Maybe add a comment that they are only loaded in the executor via the path above